akka-persistence-typed/src/main/scala/akka/persistence/typed/PersistenceId.scala

```diff
@@ -122,18 +122,28 @@ object PersistenceId {
   /**
    * Constructs a [[PersistenceId]] with `id` as the full unique identifier.
    */
-  def of(id: String): PersistenceId =
-    PersistenceId(id)
+  def ofUniqueId(id: String): PersistenceId =
+    new PersistenceId(id)
+
 }
 
 /**
  * Unique identifier in the backend data store (journal and snapshot store) of the
  * persistent actor.
  */
-final case class PersistenceId(id: String) {
+final class PersistenceId private (val id: String) {
```

Comment:
Changed to class to get rid of the apply, but also good to make it easier to evolve if we need some other change in the future.

I don't think there is a need for unapply.